### PR TITLE
Update host on Linux with Nginx

### LINKED ISSUE #5575: Update host on Linux with Nginx
The doc currently advises you to build Nginx from source if you want the http_ssl_module. Looks like that is no longer required if you install nginx from the official PPA (at least 16.04).

https://www.nginx.com/resources/wiki/start/topics/tutorials/install/#official-debian-ubuntu-packages

We should simplify the doc and no longer ask folks to build nginx from source but instead install directly from the official PPA

### LINKED ISSUE #5707: Update Nginx configuration
host-and-deploy/linux-nginx
https://docs.microsoft.com/aspnet/core/host-and-deploy/linux-nginx

References:
* https://github.com/aspnet/Home/issues/2954#issuecomment-373284921
* https://github.com/aspnet/Home/issues/2954#issuecomment-373467870
* https://github.com/yandex/gixy/blob/master/docs/en/plugins/hostspoofing.md

Fix:
> use $host instead of $http_host

cc/ @blowdart @Tratcher @shirhatti @Rick-Anderson

### LINKED ISSUE #6507: What next?
The article seems to have ended abruptly.  Does just going to the browser default path for Nginx launch the application?  Where on the Linux server do you put the published .NET Core files?  Is there a common folder?

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: dcdd2c35-bf2b-2673-d150-303a8aaa54d2
* Version Independent ID: 815fe034-c9fc-aee3-a8b5-f2e860b30302
* Content: [Host ASP.NET Core on Linux with Nginx](https://docs.microsoft.com/en-us/aspnet/core/host-and-deploy/linux-nginx?view=aspnetcore-2.0&tabs=aspnetcore2x#feedback)
* Content Source: [aspnetcore/host-and-deploy/linux-nginx.md](https://github.com/aspnet/Docs/blob/master/aspnetcore/host-and-deploy/linux-nginx.md)
* Service: **unspecified**
* Product: **asp.net-core**
* GitHub Login: @Rick-Anderson
* Microsoft Alias: **riande**

### PR DESCRIPTION
**Pull request by guardrex**
Fixes #5575 
Fixes #5707
Fixes #6507

[Internal Review Topic](https://review.docs.microsoft.com/en-us/aspnet/core/host-and-deploy/linux-nginx?branch=pr-en-us-5687)

@shirhatti I'll find some time in the next few days to run the steps and see if this actually works. lol 🤞🍀

RE: The PCRE library and the `ngx_http_rewrite_module`, should I reconstitute a note on that? It seems to me that if they were going to use that module, they would find out through `ngx_http_rewrite_module` docs that the PCRE library is required.

### CODE REVIEW

File: aspnetcore/host-and-deploy/linux-nginx.md [hunk 1]
```diff
@@ -92,14 +92,22 @@ If no [ForwardedHeadersOptions](/dotnet/api/microsoft.aspnetcore.builder.forward
 
 ### Install Nginx
 
+Use `apt-get` to install Nginx. The installer creates a System V init script that runs Nginx as daemon on system startup.
```
Comment (shirhatti) [inline aspnetcore/host-and-deploy/linux-nginx.md]: System V? Doesn't 16.04 use systemd?

Comment (guardrex) [reply aspnetcore/host-and-deploy/linux-nginx.md]: Idk ... It was a pre-existing remark in the topic ... I just moved it up.

Comment (guardrex) [reply aspnetcore/host-and-deploy/linux-nginx.md]: Seems correct ... https://www.digitalocean.com/community/tutorials/how-to-use-systemctl-to-manage-systemd-services-and-units ...

> Please note that although systemd has become the default init system for many Linux distributions ...

... they go on to say that some distros don't use it. They say to just run `systemctl` ... if it claims it isn't installed, then some other init system is in place. Of course, I'll find out when I run the steps.

**Comment by shirhatti** [comment]
Good to hear. I'd assume the https content is still accurate. I can take a quick look whenever you're ready to merge.

**Comment by guardrex** [comment]
@shirhatti Yes, please go ahead whenever ur ready and take a look.

We'll do a UE review after that and get these updates live.

The big question tho is WRT the `$http` situation. That var isn't set OOB AFAICT. I looked (briefly at 3am) and couldn't find good resources online. Perhaps, I missed something. Do you want to research that a bit more, or would you prefer that I take another look into it?

**Comment by guardrex** [comment]
@shirhatti It was a simple mistype ... "$http" for "$host" ... I just fixed it.

You can review whenever ur ready to take a look.

**Comment by guardrex** [comment]
ping @shirhatti

**Comment by Rick-Anderson** [comment]
@shirhatti Let me know when you have time for the final sign off.

File: aspnetcore/host-and-deploy/linux-nginx.md [hunk 2]
```diff
@@ -29,23 +29,40 @@ This guide:
...
 ## Prerequisites
 
-1. Access to an Ubuntu 16.04 Server with a standard user account with sudo privilege
+1. Access to an Ubuntu 16.04 server with a standard user account with sudo privilege
 1. An existing ASP.NET Core app
 
 ## Copy over the app
 
-Run [dotnet publish](/dotnet/core/tools/dotnet-publish) from the dev environment to package an app into a self-contained directory that can run on the server.
+Configure the app for a [self-contained deployment](/dotnet/core/deploying/#self-contained-deployments-scd).
```
Comment (shirhatti) [inline aspnetcore/host-and-deploy/linux-nginx.md]: I don't believe we want to actively guide customers to standalone due to concerns around servicing. Can we change this to be less opinionated?

File: aspnetcore/host-and-deploy/linux-nginx.md [hunk 3]
```diff
@@ -29,23 +29,40 @@ This guide:
...
+</PropertyGroup>
+```
+
+Run [dotnet publish](/dotnet/core/tools/dotnet-publish) from the dev environment to package an app into a self-contained directory that can run on the server:
+
+```console
+dotnet publish --configuration Release --runtime ubuntu.16.04-x64
+```
+
+Copy the ASP.NET Core app to the server using a tool that integrates into the organization's workflow (for example, SCP, FTP). It's common to locate web apps under the *var* directory (for example, *var/aspnetcore/hellomvc*).
```
Comment (shirhatti) [inline aspnetcore/host-and-deploy/linux-nginx.md]: s/FTP/SFTP

Let's not even suggest FTP

File: aspnetcore/host-and-deploy/linux-nginx.md [hunk 4]
```diff
@@ -143,6 +168,21 @@ With the preceding configuration file and default server, Nginx accepts public t
...
+To directly run the app on the server:
+
+1. Navigate to the app's directory.
+1. Run the app's executable: `./<app_executable>`.
+
+If a permissions error occurs, change the permissions:
+
+```console
+chmod u+x <app_executable>
+```
```
Comment (shirhatti) [inline aspnetcore/host-and-deploy/linux-nginx.md]: Most likely the nginx user isn't running as the current running user. `chmod u+x` wouldn't really help solve the problem. I'm not sure if `chmod +x` has any security implications? @blowdart

Comment (blowdart) [reply aspnetcore/host-and-deploy/linux-nginx.md]: I have no idea, it's linux :(

Comment (guardrex) [reply aspnetcore/host-and-deploy/linux-nginx.md]: Not sure what to do on this one @shirhatti.

**Comment by guardrex** [comment]
@shirhatti Updates made except for the last one.

Also note a few other things:
* The Linux with Apache topic is still instructing to publish an SCD. Do you want that changed, too, to a FDD?
* I see new Ubuntu runtime versions. Do you want me to try and make this more generic for the Ubuntu version? ... or is it too risky unless those versions are explicitly tested with the doc, since they might :boom: following these instructions.

**Comment by guardrex** [comment]
@Rick-Anderson @scottaddie Possible SiteHelp issue. I've seen this exact warning twice now ...

capture1

capture

... then, the final build (the one showing here now) is fine (no warning).

File: aspnetcore/host-and-deploy/linux-nginx.md (hunk 4 above)
Comment (shirhatti) [reply aspnetcore/host-and-deploy/linux-nginx.md]: Leave it as `u+x`. Less permissions is better, though it might not fix the issue being called out.

**Comment by shirhatti** [comment]
> The Linux with Apache topic is still instructing to publish an SCD. Do you want that changed, too, to a FDD?

SCP? FDD?

**Comment by guardrex** [comment]
@shirhatti Self-Contained Deployment (SCD) | Framework-Depdendent Deployment (FDD)

**Comment by guardrex** [comment]
:smile: I guess the acronyms didn't catch-on so well, did they? :trollface:

**Comment by shirhatti** [comment]
> I see new Ubuntu runtime versions. Do you want me to try and make this more generic for the Ubuntu version? ... or is it too risky unless those versions are explicitly tested with the doc, since they might 💥 following these instructions.

12.04 -> 14.04 -> 16.04 went from Sys V init to Upstart to SystemD. Though that kind of change is unlikely to happen. I'd rather specify a fixed version and say it's likely it work on newer versions but say something to the effect of it hasn't been tested on the newer versions

**Comment by guardrex** [comment]
@shirhatti See how the last commit looks.

*The Rex Legal Division has implored me to state that I ran the Nginx instructions with an SCD.* I :pray: it's all good under an FDD paradigm.

File: aspnetcore/host-and-deploy/linux-apache.md [hunk 5]
```diff
@@ -19,15 +19,29 @@ Using this guide, learn how to set up [Apache](https://httpd.apache.org/) as a r
...
+   1. Select and follow the instructions for CentOS/Oracle.
+1. An existing ASP.NET Core app.
 
-## Publish the app
+## Publish and copy over the app
 
-Publish the app as a [self-contained deployment](/dotnet/core/deploying/#self-contained-deployments-scd) in Release configuration for the CentOS 7 runtime (`centos.7-x64`). Copy the contents of the *bin/Release/netcoreapp2.0/centos.7-x64/publish* folder to the server using SCP, FTP, or other file transfer method.
+Configure the app for a [framework-dependent deployment](/dotnet/core/deploying/#framework-dependent-deployments-fdd).
+
+Run [dotnet publish](/dotnet/core/tools/dotnet-publish) from the development environment to package an app into a directory (*bin/Release/netcoreapp2.x/publish*) that can run on the server:
```
Comment (scottaddie) [inline aspnetcore/host-and-deploy/linux-apache.md]: It's probably best not to use "netcoreapp2.x" in this sample path. What about using a placeholder instead? Maybe something like *<target_framework_moniker>*. I also recommend prefixing the path with something like "for example,".

File: aspnetcore/host-and-deploy/linux-nginx.md [hunk 6]
```diff
@@ -29,23 +29,40 @@ This guide:
...
+1. An existing ASP.NET Core app.
 
-## Copy over the app
+## Publish and copy over the app
 
-Run [dotnet publish](/dotnet/core/tools/dotnet-publish) from the dev environment to package an app into a self-contained directory that can run on the server.
+Configure the app for a [framework-dependent deployment](/dotnet/core/deploying/#framework-dependent-deployments-fdd).
 
-Copy the ASP.NET Core app to the server using whatever tool integrates into the organization's workflow (for example, SCP, FTP). Test the app, for example:
+Run [dotnet publish](/dotnet/core/tools/dotnet-publish) from the development environment to package an app into a directory (*bin/Release/netcoreapp2.x/publish*) that can run on the server:
```
Comment (scottaddie) [inline aspnetcore/host-and-deploy/linux-nginx.md]: I have the same recommendation with this path.

**Comment by guardrex** [comment]
@scottaddie Updates made. :+1:

**Pull request merged**